lib/engine/game/g_18_esp/step/choose.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G18ESP
+      module Step
+        class Choose < Engine::Step::Base
+          def description
+            'Choose a Train'
+          end
+
+          def actions(entity)
+            actions = []
+            actions << 'choose' if @game.p2&.owner == entity
+
+            actions
+          end
+
+          def log_skip(_entity); end
+
+          def choice_name
+            'Choose train'
+          end
+
+          def choices
+            { '2': '2', '1+2': '1+2' }
+          end
+
+          def active_entities
+            [@game.p2&.owner] if @game.p2&.owner&.corporation?
```

Comment:
I think you need a return statement here. Otherwise this method is always returning the value from the superclass's method.
```suggestion
            return [@game.p2&.owner] if @game.p2&.owner&.corporation?
```

Reply:
didn't need active_entities redefined here, so i just removed this section